Build fails on physfs:

On Arch Linux with gcc 14.1.1 the build is failing on physfs. When I build physfs alone from it's repo on the same commit it works fine.

```
make -j5  
[  1%] Building C object CMakeFiles/c-hashmap-static.dir/home/family/projects/VVVVVV/third_party/c-hashmap/map.c.o
[  2%] Building C object CMakeFiles/faudio-static.dir/home/family/projects/VVVVVV/third_party/FAudio/src/FAudio.c.o
[  3%] Building C object CMakeFiles/lodepng-static.dir/src/lodepng_wrapper.c.o
[  4%] Building C object CMakeFiles/sheenbidi-static.dir/home/family/projects/VVVVVV/third_party/SheenBidi/Source/SheenBidi.c.o
-- This is interim commit 16d75d2d (committed 2024-05-25) on branch master
[  4%] Built target GenerateVersion
[  6%] Building C object CMakeFiles/faudio-static.dir/home/family/projects/VVVVVV/third_party/FAudio/src/FAudio_internal.c.o
[  7%] Linking C static library libc-hashmap-static.a
[  7%] Built target c-hashmap-static
[  8%] Building C object CMakeFiles/faudio-static.dir/home/family/projects/VVVVVV/third_party/FAudio/src/FAudio_internal_simd.c.o
[  9%] Building CXX object CMakeFiles/tinyxml2-static.dir/home/family/projects/VVVVVV/third_party/tinyxml2/tinyxml2.cpp.o
[ 11%] Building C object CMakeFiles/faudio-static.dir/home/family/projects/VVVVVV/third_party/FAudio/src/FAudio_operationset.c.o
[ 12%] Building C object CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs.c.o
[ 13%] Linking C static library libsheenbidi-static.a
[ 13%] Built target sheenbidi-static
[ 14%] Building C object CMakeFiles/faudio-static.dir/home/family/projects/VVVVVV/third_party/FAudio/src/FAudio_platform_sdl2.c.o
[ 16%] Building C object CMakeFiles/InterimVersion.dir/src/InterimVersion.out.c.o
[ 17%] Linking C static library libInterimVersion.a
[ 17%] Built target InterimVersion
[ 18%] Building C object CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_archiver_dir.c.o
[ 20%] Building C object CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_archiver_unpacked.c.o
[ 20%] Linking C static library liblodepng-static.a
[ 22%] Building C object CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_archiver_zip.c.o
[ 23%] Building C object CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_byteorder.c.o
[ 24%] Linking CXX static library libtinyxml2-static.a
[ 24%] Built target lodepng-static
[ 25%] Building C object CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_unicode.c.o
[ 25%] Built target tinyxml2-static
[ 27%] Building C object CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_platform_unix.c.o
[ 28%] Building C object CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_platform_posix.c.o
/home/family/projects/VVVVVV/third_party/physfs/src/physfs_platform_unix.c: In function ‘readSymLink’:
/home/family/projects/VVVVVV/third_party/physfs/src/physfs_platform_unix.c:230:15: error: implicit declaration of function ‘readlink’ [-Wimplicit-function-declaration]
  230 |          rc = readlink(path, retval, len);
      |               ^~~~~~~~
make[2]: *** [CMakeFiles/physfs-static.dir/build.make:174: CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_platform_unix.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/home/family/projects/VVVVVV/third_party/physfs/src/physfs_platform_posix.c: In function ‘__PHYSFS_platformStat’:
/home/family/projects/VVVVVV/third_party/physfs/src/physfs_platform_posix.c:334:53: error: implicit declaration of function ‘lstat’; did you mean ‘fstat’? [-Wimplicit-function-declaration]
  334 |     const int rc = follow ? stat(fname, &statbuf) : lstat(fname, &statbuf);
      |                                                     ^~~~~
      |                                                     fstat
make[2]: *** [CMakeFiles/physfs-static.dir/build.make:160: CMakeFiles/physfs-static.dir/home/family/projects/VVVVVV/third_party/physfs/src/physfs_platform_posix.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:288: CMakeFiles/physfs-static.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 29%] Linking C static library libfaudio-static.a
[ 29%] Built target faudio-static
make: *** [Makefile:91: all] Error 2
```

--- a/desktop_version/CMakeLists.txt
+++ b/desktop_version/CMakeLists.txt
@@ -65,7 +65,7 @@ if(NOT WIN32)
 endif()
 
 # Source Lists
-set(VVV_SRC
+set(VVV_CXX_SRC
     src/BinaryBlob.cpp
     src/BlockV.cpp
     src/ButtonGlyphs.cpp
@@ -108,6 +108,8 @@ set(VVV_SRC
     src/WarpClass.cpp
     src/XMLUtils.cpp
     src/main.cpp
+)
+set(VVV_C_SRC
     src/DeferCallbacks.c
     src/GlitchrunnerMode.c
     src/Network.c
@@ -120,11 +122,13 @@ set(VVV_SRC
     ../third_party/physfs/extras/physfsrwops.c
 )
 if(STEAM)
-    list(APPEND VVV_SRC src/SteamNetwork.c)
+    list(APPEND VVV_C_SRC src/SteamNetwork.c)
 endif()
 if(GOG)
-    list(APPEND VVV_SRC src/GOGNetwork.c)
+    list(APPEND VVV_C_SRC src/GOGNetwork.c)
 endif()
+
+set(VVV_SRC ${VVV_CXX_SRC} ${VVV_C_SRC})
 
 # Executable information
 if(WIN32)
@@ -290,11 +294,11 @@ if(MSVC)
     # MSVC does not officially support disabling exceptions,
     # so this is as far as we are willing to go to disable them.
     string(REGEX REPLACE "/EH[a-z]+" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /EHsc")
+    set_source_files_properties(${VVV_CXX_SRC} PROPERTIES COMPILE_FLAGS /EHsc)
 
     # Disable RTTI
     string(REPLACE "/GR" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /GR-")
+    set_source_files_properties(${VVV_CXX_SRC} PROPERTIES COMPILE_FLAGS /GR-)
 
     if(MSVC_VERSION GREATER 1900)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /utf-8")
@@ -302,18 +306,18 @@ if(MSVC)
     endif()
 else()
     string(REGEX REPLACE "-std=[a-z0-9]+" "" CMAKE_C_FLAGS "${CMAKE_C_FLAGS}")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99")
+    set_source_files_properties(${VVV_C_SRC} PROPERTIES COMPILE_FLAGS -std=c99)
 
     string(REGEX REPLACE "-std=[a-z0-9+]+" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++98")
+    set_source_files_properties(${VVV_CXX_SRC} PROPERTIES COMPILE_FLAGS -std=c++98)
 
     # Disable exceptions
     string(REPLACE "-fexceptions" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-exceptions")
+    set_source_files_properties(${VVV_CXX_SRC} PROPERTIES COMPILE_FLAGS -fno-exceptions)
 
     # Disable RTTI
     string(REPLACE "-frtti" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-rtti")
+    set_source_files_properties(${VVV_CXX_SRC} PROPERTIES COMPILE_FLAGS -fno-rtti)
 endif()
 
 # Unfortunately, it doesn't seem like distros package LodePNG

--- a/desktop_version/CMakeLists.txt
+++ b/desktop_version/CMakeLists.txt
@@ -318,6 +318,10 @@ else()
     # Disable RTTI
     string(REPLACE "-frtti" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
     set_source_files_properties(${VVV_CXX_SRC} PROPERTIES COMPILE_FLAGS -fno-rtti)
+
+    # Dependencies (as needed)
+    set_source_files_properties(${FAUDIO_SRC} PROPERTIES COMPILE_FLAGS -std=c99)
+    set_source_files_properties(${CHM_SRC} PROPERTIES COMPILE_FLAGS -std=c99)
 endif()
 
 # Unfortunately, it doesn't seem like distros package LodePNG

